routing/payment_session.go

```diff
@@ -154,6 +158,13 @@ func (p *paymentSession) RequestRoute(payment *LightningPayment,
 		return nil, fmt.Errorf("pre-built route already tried")
 	}
 
+	// Add BlockPadding to the finalCltvDelta so that the receiving node
+	// does not reject the HTLC if a block is mined while its in-flight.
+	// This is only added if no finalCltvDelta was set.
+	if finalCltvDelta == 0 {
```

Comment:
If we only add block padding in this case, then it doesn't address the common case where a users payment fails due to a block being mined _right_ as they send. We _always_ set our `finalCltvDelta` if the payment goes through `SendPayment`.

Reply:
Ahhhh! I missed that.  Joost had a comment above where he said the block padding should *only* be specified iff no `finalCltvDelta` was supplied.  But it seems that has some problems as well.
see: https://github.com/lightningnetwork/lnd/pull/3153#discussion_r302687572

Reply:
From my reading, his comment only applied to `queryroutes`. IMO, we should also apply the padding there, as users are known to execute commands that resemble: `queryroutes | sendtoroute`.

Reply:
Yeah that's true, should it be set any time a route is requested then or perhaps this could be the default with an option to turn the padding off?

Reply:
Another way to see this is, is from the perspective of keeping the api surface as small as possible. `QueryRoutes` (the rpc) is for advanced api users and you could also say that they should just supply just an absolute block height. This should include the final cltv expiry of the invoice and block padding. It would give more control over the time locks of the route returned, which may be required if there is chaining involved with time locks of other payments.

`QueryRoutes` (the rpc) is different from `queryroutes` (the cli command). We could choose to have `lncli` do some of the calculation to still be able to `queryroutes | sendtoroute`.